### PR TITLE
Feature/20 date picker and date handling

### LINKED ISSUE #20: Date picker and date handling
- [x] Add date picker to search page
- [x] Always fetch resources with date info (default to current date)
- [x] Always fetch resource on resource page with date info (default to current date)

### PR DESCRIPTION
**Pull request by lraivio**
This PR:
- Adds DatePicker component
- Adds date to search controls
- Always fetches resources with start and end times.

Closes #20.

### CODE REVIEW

File: app/state/reducers/searchFiltersReducer.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
+import moment from 'moment';
 import { handleActions } from 'redux-actions';
 
-const initialState = {
-  search: '',
-};
+function getInitialState() {
+  return {
+    date: moment().format('YYYY-MM-DD'),
```
Comment (FelixLoether) [inline app/state/reducers/searchFiltersReducer.js]: Do you think we should always save dates as strings in the store? And in general where do you think we should be using Moments/strings in the code?

Comment (lraivio) [inline app/state/reducers/searchFiltersReducer.js]: That's a good question. My initial thought was to always store dates as strings and only change them to moment objects where needed. Makes working with url changes and filtering easier but might make other stuff harder.
One thing to also think about is where to store the format info. The 'YYYY-MM-DD' format is for example repeated in lot of places.

File: app/api/actions/resources.js [hunk 2]
```diff
@@ -23,7 +21,7 @@ function fetchResource(id, params = {}) {
 function fetchResources(params = {}) {
   return createApiAction({
     endpoint: 'resource',
-    params: Object.assign({}, params, { pageSize: 100 }),
+    params: { ...getParamsWithTimes(params), ...{ pageSize: 100 } },
```
Comment (FelixLoether) [inline app/api/actions/resources.js]: - [ ] The ... is not required for the second one. `{ ...getParamsWithTimes(params), pageSize: 100 }`.

File: app/shared/date-picker/DatePicker.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+function DatePicker(props) {
+  return (
+    <DateField
+      className="date-picker"
+      clearIcon={false}
+      collapseOnDateClick
+      dateFormat={localizedDateFormat}
+      footer={false}
+      onChange={date => props.onChange({ date: formatDate(date) })}
```
Comment (FelixLoether) [inline app/shared/date-picker/DatePicker.js]: - [ ] If I was using this component I'd expect the argument to onChange to be either the change event or the new value, not necessarily an object that contains the new value in a `date` attribute. So I think it'd make more sense for this to be `onChange={props.onChange}` or `onChange={date => props.onChange(formatDate(date))}`.

**Comment by FelixLoether** [comment]
These changes are not necessary for the demo, so this is good to merge to a demo branch if necessary.

File: app/shared/date-picker/DatePicker.js (hunk 3 above)
Comment (lraivio) [inline app/shared/date-picker/DatePicker.js]: Good point 👍  
Especially since this is a shared component.

**Comment by lraivio** [comment]
Changes made.

**Comment by FelixLoether** [comment]
You can rebase + merge.

**Pull request merged**